python3/packages/observer.py

```diff
@@ -333,16 +333,17 @@ def autoinstrument_class(aclass):
         def autoinstrument_module(amodule):
             """Autoinstrument the classes and functions in a module."""
 
-            # Instrument the methods of the classes in the module
-            for _, aclass in inspect.getmembers(amodule, inspect.isclass):
-                try:
-                    autoinstrument_class(aclass)
-                except Exception:
-                    debug("instrument_function: Exception %s", traceback.format_exc())
-
-            # Instrument the module-level functions of the module
-            for fname, afunction in inspect.getmembers(amodule, inspect.isfunction):
-                setattr(amodule, fname, instrument_function(afunction))
+            with tracers[0].start_as_current_span(f"auto_instrumentation.add_module: {amodule}"):
```

Comment:
From just looking at the diff (without reading the PR description) it wasn't easy to see to me what the change does. I think that this is an opportunity to add a comment which may explain why the context manager is added:

```suggestion
            # Group the aut_instrumentation spans by module:
            # Add a parent span for the auto_instrumentation of each module to group
            # them together so that they can be hidden more easily in Jaeger.
            with tracers[0].start_as_current_span(f"auto_instrumentation.add_module: {amodule}"):
```
This is only a proposal, but think one to three lines of comments should be fine here.

Reply:
Just the code as written: it's creating a span called "auto_instrumentation.add_module" which I think is explained by its name. See the counterparts that are already present, "auto_instrumentation.add_class" and "class.instrument". I think some understanding of distributed tracing is expected (which is explained elsewhere in the script)

Hiding them in Jaeger is one aspect but it should have been present to begin with really, for the same reason add_class and class.instrument already exist.